src/ptvsd/adapter/ide.py

```diff
@@ -232,6 +232,10 @@ def attach_request(self, request):
         if self.session.no_debug:
             raise request.isnt_valid('"noDebug" is not supported for "attach"')
 
+        if self.server is not None:
+            # we are already connected to the debug server
+            return
+
```

Comment:
This should go into the first branch below, and the PID branch should check for `self.server is not None` and fail the request if so.